src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -427,5 +528,194 @@ private ImmutableArray<BoundExpression> BindInterpolatedStringParts(BoundUnconve
+                return CreateConversion(
+                    unconvertedString.Syntax,
+                    unconvertedString,
+                    interpolatedStringConversion,
+                    isCast: false,
+                    conversionGroupOpt: null,
+                    wasCompilerGenerated: false,
+                    interpolatedStringParameter.Type,
+                    diagnostics,
+                    hasErrors: true);
+            }
+
+            var handlerParameterIndexes = interpolatedStringParameter.InterpolatedStringHandlerArgumentIndexes;
+            if (handlerParameterIndexes.IsEmpty)
+            {
+                // No arguments, fall back to the standard conversion steps.
+                return CreateConversion(
+                    unconvertedString.Syntax,
+                    unconvertedString,
+                    interpolatedStringConversion,
+                    isCast: false,
+                    conversionGroupOpt: null,
+                    interpolatedStringParameter.Type,
+                    diagnostics);
+            }
+
+            Debug.Assert(handlerParameterIndexes.All((index, paramLength) => index >= BoundInterpolatedStringArgumentPlaceholder.InstanceParameter && index < paramLength,
+                                                     parameters.Length));
+
+            // We need to find the appropriate argument expression for every expected parameter, and error on any that occur after the current parameter
+
+            ImmutableArray<int> handlerArgumentIndexes;
+
+            if (memberAnalysisResult.ArgsToParamsOpt.IsDefault && arguments.Count == parameters.Length)
+            {
+                // No parameters are missing and no remapped indexes, we can just use the original indexes
+                handlerArgumentIndexes = handlerParameterIndexes;
+            }
+            else
+            {
+                // Args and parameters were reordered via named parameters, or parameters are missing. Find the correct argument index for each parameter.
+                var handlerArgumentIndexesBuilder = ArrayBuilder<int>.GetInstance(handlerParameterIndexes.Length, fillWithValue: BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter);
+                for (int handlerParameterIndex = 0; handlerParameterIndex < handlerParameterIndexes.Length; handlerParameterIndex++)
+                {
+                    int handlerParameter = handlerParameterIndexes[handlerParameterIndex];
+                    Debug.Assert(handlerArgumentIndexesBuilder[handlerParameterIndex] is BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter);
+
+                    if (handlerParameter == BoundInterpolatedStringArgumentPlaceholder.InstanceParameter)
+                    {
+                        handlerArgumentIndexesBuilder[handlerParameterIndex] = handlerParameter;
```

Comment:
>handlerArgumentIndexesBuilder[handlerParameterIndex] = handlerParameter; [](http://example.com/codeflow?start=24&length=72)

What unit-test covers this fix?

Reply:
`InterpolatedStringHandlerArgumentAttribute_ParametersReordered` threw before this fix.